ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java

```diff
@@ -573,6 +575,65 @@ public static void processExits(BeaconState state, BeaconBlock block)
     }
   }
 
+  /**
+   * @param state
+   * @param block
+   * @see
+   *     https://github.com/ethereum/eth2.0-specs/blob/v0.1/specs/core/0_beacon-chain.md#transfers-1
+   */
+  public static void processTransfers(BeaconState state, BeaconBlock block)
+      throws BlockProcessingException {
+    // Verify that len(block.body.exits) <= MAX_EXITS
+    checkArgument(block.getBody().getTransfers().size() <= Constants.MAX_TRANSFERS);
+    checkArgument(allDistinct(block.getBody().getTransfers()));
```

Comment:
This can also be done with streams without introducing a separate method. Something like:
```
checkArgument(block.getBody().getTransfers().stream().distinct().collect(Collectors.toList()).size() == block.getBody().getTransfers().size());
```
(untested!)

On the happy flow, we need to iterate over all elements anyway, so there's no performance loss by not exiting early. On the unhappy flow, we don't really care about performance, I guess (so early exit does not matter).

I'll wait for a Java expert to come along and say about which is preferred 😄

Reply:
Hmm, I'm thinking about this now while giving the processTransfers logic a double check.

Reply:
The separate method does serve as documentation, though, so that may be preferable.

Reply:
Yeah I think I've decided on the separate method being fine, for both documentation and maintainability. I'm sure we'll revisit this in our one of many refactors down the line if it ends up bugging someone.